Doc hydro signatures (#373)
DassHydro/smash: doc/source/user_guide/classical_uses/hydrological_signatures.rst

the flood event runoff coefficient (``'Erc'``), and the peak flow (``'Epf'``). 

.. ipython:: python

    sig_obs = smash.signatures(model, sign=["Crc", "Cfp90", "Erc", "Epf"], domain="obs")

To compute the simulated signatures, a simulation (either forward run or optimization) has to be performed to generate the simulated discharge.
We compute the same signatures as the observed ones for the simulated discharge.

.. ipython:: python

    model.forward_run()

    sig_sim = smash.signatures(model, sign=["Crc", "Cfp90", "Erc", "Epf"], domain="sim")

The signatures computation result object contains two attributes which are two different dictionaries:

- ``cont`` : Continuous signatures computation result,

- ``event``: Flood event signatures computation result.

.. ipython:: python

    sig_sim.cont

    sig_sim.event

For flood event signatures computation, more options can be specified such as the threshold for flood event detection, the maximum duration of the flood event, etc.
The segmentation algorithm used to detect the flood events can be adjusted by setting the ``event_seg`` parameter in the `smash.signatures` function.

> This parameter is a dictionary with keys that are the parameters used for the segmentation algorithm (refer to the tutorial on :ref:`hydrograph segmentation <user_guide.classical_uses.hydrograph_segmentation>` for more details).

Author: let's try that

For instance, we can reduce the quantile threshold for flood event detection to 0.99.

.. ipython:: python

    sig_obs_2 = smash.signatures(model, sign=["Erc", "Epf"], domain="obs", event_seg={"peak_quant": 0.99})
    sig_obs_2.event

    sig_sim_2 = smash.signatures(model, sign=["Erc", "Epf"], domain="sim", event_seg={"peak_quant": 0.99})
    sig_sim_2.event

Signatures visualization
************************

Now, we visualize, for instance, the simulated and observed peak flow in the boxplot below.

.. ipython:: python

    df_obs = sig_obs_2.event
    df_sim = sig_sim_2.event

> 

Author: ratafia

Hydrological Signatures
=======================

This section aims to go into detail on how to compute and vizualize some hydrological signatures
(a list of studied signatures and their formula is given in :ref:`Math / Num Documentation <math_num_documentation.hydrological_signature>`).

First, open a Python interface:

.. code-block:: none

    python3

Imports
*******

We will first import everything we need in this tutorial

.. ipython:: python

    import smash
    import pandas as pd
    import matplotlib.pyplot as plt

Model object creation
*********************

Now, we need to create a :class:`smash.Model` object.
For this case, we will use the :ref:`user_guide.demo_data.cance` dataset as an example.

Load the ``setup`` and ``mesh`` dictionaries using the `smash.factory.load_dataset` function and create the :class:`smash.Model` object.

.. ipython:: python

    setup, mesh = smash.factory.load_dataset("Cance")
    
    model = smash.Model(setup, mesh)

Signatures computation
**********************

> To start we need to do a forward_run (or  optimization) simulation. Then the sinatures computation is available using the `smash.signatures` function which has three argument (model, domain and event_seg).

Reviewer: The `smash.signatures` function provide to compute signatures.
By default, the signatures are computed on the observed informations of hydrograph segmentation of flood events of the model, but they could be based on the simulated data, by changing the argument ``domain`` of the function. About the default definition of hydrograph segmentation, see :ref:`user_guide.classical_uses.hydrograph_segmentation`.

.. ipython:: python

    model.forward_run()

>     sig_obs = smash.signatures(model, domain="obs")

Reviewer: ```suggestion
    sig_obs = smash.signatures(model)
```

    sig_sim = smash.signatures(model, domain="sim")

> The signatures coputation result object contains 2 attributes which are 2 different dictionaries:

Reviewer: ```suggestion
The signatures computation result object contains two attributes which are two different dictionaries ``cont`` and ``event``:
```

> - ``cont`` : Continuous signatures computation result,

Reviewer: ```suggestion
- ``cont`` : Continuous signatures computation result, on all the discharge time series,
```

> - ``event``: Flood event signatures computation result.

Reviewer: ```suggestion
- ``event``: Flood events signatures computation result.
```

> For example, we can display the simulated continuous and flood event signatures as follow.

Reviewer: ```suggestion
We can display the simulated continuous and flood events signatures as follow.
```

.. ipython:: python

    sig_sim.cont

    sig_sim.event

> Now, we visualize, for instance, the simulated and observed flood event runoff coefficients in the boxplots below.

Reviewer: ```suggestion
Now, we visualize the boxplots of simulated and observed flood event runoff coefficients as below.
```

.. ipython:: python

    df_obs = sig_obs.event
    df_sim = sig_sim.event

    df = pd.concat([df_obs, df_sim], ignore_index=True)
    df["domain"] = ["obs"]*len(df_obs) + ["sim"]*len(df_sim)

    @savefig user_guide.classical_uses.hydrological_signatures_boxplot.png
    boxplot = df.boxplot(column=["Erc", "Erchf", "Erclf", "Erch2r"], by="domain")

> 

Reviewer: Can u add a qualitative explanation about each signature u show ?
